site/public/index.php

```diff
@@ -63,20 +60,27 @@ function error_handler() {
         " . $error['file'] . " on line " . $error['line']));
     }
 }
+
 register_shutdown_function("error_handler");
+
 /*
  * Check that we have a semester and a course specified by the user and then that there's no
  * potential for path trickery by using basename which will return only the last part of a
  * given path (such that /../../test would become just test)
  */
 if (!isset($_REQUEST['semester'])) {
-    // @todo: should check for a default semester if one is not specified, opposed to throwing an exception
-    $core->getOutput()->showError("Need to specify a semester in the URL");
+    // @todo: should check for a default semester if one is not specified, rather than leaving empty.
+    $_REQUEST['semester'] = "";
 }
 if (!isset($_REQUEST['course'])) {
-    $core->getOutput()->showError("Need to specify a course in the URL");
+    $_REQUEST['course'] = "";
+}
+//For now, if either semester or course is blank, we will assume both to be. @todo Handle this more elegantly.
+if($_REQUEST['semester'] == "" || $_REQUEST['course'] == ""){
+    $_REQUEST['semester'] = $_REQUEST['course'] = "";
 }
```

Comment:
We can probably just simply the above checks to just check for [empty](http://php.net/manual/en/function.empty.php) which would cover both a check against `== ""` and `isset` (`empty($_REQUEST['semester']) || empty($_REQUEST['course'])`).